Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -618,19 +682,17 @@ CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
 }
 
 /**
- * Private helper that parses a font name, and returns appropriate weight, stretch, and style values for DWrite accordingly
+ * Private helper that parses a font name, and returns appropriate weight, stretch, style, and family name values
  */
-static void __InitDWriteFontPropertiesFromName(CFStringRef fontName,
-                                               DWRITE_FONT_WEIGHT* weight,
-                                               DWRITE_FONT_STRETCH* stretch,
-                                               DWRITE_FONT_STYLE* style) {
+static void __InitDWriteFontPropertiesFromName(
+    CFStringRef fontName, DWRITE_FONT_WEIGHT* weight, DWRITE_FONT_STRETCH* stretch, DWRITE_FONT_STYLE* style, CFStringRef* familyName) {
     // Set some defaults for when weight/stretch/style are not mentioned in the name
     // Since this is a file-private helper, assume the pointers are safe to dereference.
     *weight = DWRITE_FONT_WEIGHT_NORMAL;
     *stretch = DWRITE_FONT_STRETCH_NORMAL;
     *style = DWRITE_FONT_STYLE_NORMAL;
 
-    CFStringRef familyName = _DWriteGetFamilyNameForFontName(fontName);
+    *familyName = _DWriteGetFamilyNameForFontName(fontName);
```

Comment:
I mean, this currently works just fine because the string is being held in a dictionary and thus is guaranteed to last for as long as we need it, and has balanced retain count. Is your concern just that if we didn't know about that implementation detail, we wouldn't be able to guarantee that this'd be the case, because autorelease wouldn't do the correct thing in this situation?

---
In reply to: [79964653](https://github.com/Microsoft/WinObjC/pull/1021#discussion_r79964653) [](ancestors = 79964653)